### PR TITLE
Copy editing of vignettes/examples

### PR DESCRIPTION
**Pull request by kevinykuo**
Copy-editing vignettes and fix a couple bugs/regressions in examples.

### CODE REVIEW

File: index.Rmd [hunk 1]
```diff
@@ -81,12 +81,12 @@ Here, we define a helper function that will return an input function for a subse
 library(tfestimators)
 
 # return an input_fn for a given subset of data
-mtcars_input_fn <- function(data) {
+mtcars_input_fn <- function(data, num_epochs = 1) {
   input_fn(data, 
            features = c("disp", "cyl"), 
            response = "mpg",
            batch_size = 32,
-           num_epochs = 10)
```
Comment (kevinykuo) [inline index.Rmd]: Doesn't make sense to set epochs > 1 for evaluation

File: vignettes/creating_estimators.Rmd [hunk 2]
```diff
@@ -126,12 +126,12 @@ predict_data <- downloaded_data$predict_data
 
 ```
 
-Next, we construct the input function like the following:
+Next, we construct the input function as follows:
 
 ```r
 
 constructed_input_fn <- function(dataset) {
-  input_fn(dataset, features = -num_rings, response = num_rings, num_epochs = NULL, shuffle = TRUE)
```
Comment (kevinykuo) [inline vignettes/creating_estimators.Rmd]: Let `input_fn()` resolve `shuffle` automatically

File: vignettes/creating_estimators.Rmd [hunk 3]
```diff
@@ -441,7 +437,6 @@ model_fn <- function(features, labels, mode, params, config) {
   output_layer <- tf$layers$dense(second_hidden_layer, 1L)
   
   # Reshape output layer to 1-dim Tensor to return predictions
-  # TODO: This failed if it's c(-1L) - check in reticulate for single element vector conversion
```
Comment (kevinykuo) [inline vignettes/creating_estimators.Rmd]: Expected behavior, length one vectors are mapped to scalars

Comment (jjallaire) [reply vignettes/creating_estimators.Rmd]: I just re-exported the `shape` function from tensorflow, which will enable us to write this as:

```r
predictions <- tf$reshape(output_layer, shape(-1))
```

Comment (kevinykuo) [reply vignettes/creating_estimators.Rmd]: Should we discourage folks from writing things like `list(-1)` or `list(NULL, 3)` and encourage `shape()` across the board?

Comment (jjallaire) [reply vignettes/creating_estimators.Rmd]: This PR looks good to me, feel free to merge when ready!

Comment (jjallaire) [reply vignettes/creating_estimators.Rmd]: I am torn on this as `list` is a bit more transparent. 

BTW we should *never* require the `L` suffix in our R facing APIs. I realize in this case we are calling a `tf$` API which requires the `L` (so `shape` is handy because it does the as.integer).
